src-plugins/itkDataImage/writers/itkDicomDataImageWriter.cpp

```diff
@@ -0,0 +1,492 @@
+/*=========================================================================
+
+ medInria
+
+ Copyright (c) INRIA 2013 - 2014. All rights reserved.
+ See LICENSE.txt for details.
+
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#include <itkDicomDataImageWriter.h>
+
+#include <medAbstractData.h>
+#include <medAbstractDataFactory.h>
+
+
+#include <itkExtractImageFilter.h>
+#include <itkGDCMImageIO.h>
+#include <itkImage.h>
+#include <itkImageFileWriter.h>
+#include <itkMetaDataObject.h>
+
+#include <gdcmUIDGenerator.h>
+
+#include <string>
+
+static QString s_identifier() {
+    return "itkDicomDataImageWriter";
+}
+
+static QStringList s_handled() {
+    return QStringList()  << "itkDataImageChar3"
+                          << "itkDataImageUChar3"
+                          << "itkDataImageShort3"
+                          << "itkDataImageUShort3"
+                          << "itkDataImageInt3"
+                          << "itkDataImageUInt3"
+                          << "itkDataImageLong3"
+                          << "itkDataImageULong3"
+                          << "itkDataImageFloat3"
+                          << "itkDataImageDouble3";
+}
+
+itkDicomDataImageWriter::itkDicomDataImageWriter(): itkDataImageWriterBase() {
+    this->io = itk::GDCMImageIO::New();
+}
+
+itkDicomDataImageWriter::~itkDicomDataImageWriter()
+{
+}
+
+QStringList itkDicomDataImageWriter::handled() const
+{
+    return s_handled();
+}
+
+QStringList itkDicomDataImageWriter::supportedFileExtensions() const
+{
+    return QStringList() << ".dcm";
+}
+
+bool itkDicomDataImageWriter::registered()
+{
+    return medAbstractDataFactory::instance()->registerDataWriterType(s_identifier(), s_handled(), create);
+}
+
+QString itkDicomDataImageWriter::identifier() const
+{
+    return s_identifier();
+}
+
+QString itkDicomDataImageWriter::description() const
+{
+    return "Dicom image exporter";
+}
+
+QString itkDicomDataImageWriter::sopClassUID(QString modality)
+{
+    if( modality == QString("CT"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.2";
+    }
+    else if( modality == QString("MR"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.4";
+    }
+    else if( modality == QString("CR"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.1";
+    }
+    else if( modality == QString("MN"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.20";
+    }
+    else if( modality == QString("US"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.6.1";
+    }
+    else
+    {
+        qDebug( "Unknown modality. Using CT Image Storage SOP class UID.");
+        return "1.2.840.10008.5.1.4.1.1.2";
+    }
+}
+
+template <class PixelType> bool itkDicomDataImageWriter::writeDicom(const QString &path)
```

Comment:
This is a veryyy long function. It could be interesting to separate it into several smaller ones